src/registry/mod.rs

```diff
@@ -75,7 +192,16 @@ impl FromStr for RegistryUri {
     }
 }
 
-fn sanity_check_url(url: &Url) -> miette::Result<()> {
+/// Ensure that the URL is valid for a registry
+///
+/// A valid registry URL must:
+/// - Have a scheme of either "http" or "https"
+/// - End with "/artifactory" if the host is a JFrog Artifactory instance
+/// - Have a host component
+///
+/// # Arguments
+/// * `url` - The URL to check
+pub fn sanity_check_url(url: &Url) -> miette::Result<()> {
```

Comment:
Please don't make this part of the public api

Reply:
a2dc4e00eda024debb7d9bebe0ebb7ece03287f7